lib/utils.ml

```diff
@@ -47,6 +47,19 @@ let process_slack_event (ctx : Context.t) headers body ~event_handler =
     event_handler notification.event
   )
 
+(** [ process_slack_interaction ] handles slack interactions which are
+    similar to slack notifications except that they are specifically
+    for handling features such as block actions, shortcuts and modals
+*)
+let process_slack_interaction (ctx : Context.t) headers body ~interaction_handler =
+  let payload = Uri.query_of_encoded body |> List.assoc "payload" |> List.hd in
```

Comment:
List.hd is a code smell. It's better to use a match and fail with a clear message if the list is empty. It also becomes self documented this way